integration_test/cases/repo.exs

```diff
@@ -969,6 +969,80 @@ defmodule Ecto.Integration.RepoTest do
       source = from p in Permalink, select: %{url: p.title, post_id: ^post_id}
       assert {1, _} = TestRepo.insert_all(Permalink, source)
     end
+
+    @tag :upsert_all
+    @tag :with_conflict_target
+    @tag :returning
+    test "insert_all with query and source" do
```

Comment:
Is this feature related at all to conflict targets? They are orthogonal no? Could it be written in a simpler style, as the test above, or is it harder to test without a upsert?